awx/ui_next/src/screens/InstanceGroup/InstanceGroupList/InstanceGroupList.jsx

```diff
@@ -1,14 +1,209 @@
-import React from 'react';
+import React, { useEffect, useCallback } from 'react';
+import { useLocation, useRouteMatch } from 'react-router-dom';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
 import { Card, PageSection } from '@patternfly/react-core';
 
-function InstanceGroupList() {
+import { InstanceGroupsAPI } from '../../../api';
+import { getQSConfig, parseQueryString } from '../../../util/qs';
+import useRequest, { useDeleteItems } from '../../../util/useRequest';
+import useSelected from '../../../util/useSelected';
+import PaginatedDataList, {
+  ToolbarDeleteButton,
+  ToolbarAddButton,
+} from '../../../components/PaginatedDataList';
+import ErrorDetail from '../../../components/ErrorDetail';
+import AlertModal from '../../../components/AlertModal';
+import DatalistToolbar from '../../../components/DataListToolbar';
+import InstanceGroupListItem from './InstanceGroupListItem';
+
+const QS_CONFIG = getQSConfig('instance_group', {
+  page: 1,
+  page_size: 20,
+});
+
+const modifyInstanceGroups = (items = []) => {
```

Comment:
You _could_ do this work inside the fetch request which would allow you to remove this block of code and that code on line 73.  I think doing that would make it a bit easier to maintain this file in the future as you won't have to move up and down the file to remember whats happening with the instance groups.

Reply:
I tried that one before it was convoluted.